models/issues/pull.go

```diff
@@ -1093,3 +1093,27 @@ func InsertPullRequests(ctx context.Context, prs ...*PullRequest) error {
 	}
 	return committer.Commit()
 }
+
+// GetPullRequestByMergedCommit returns a merged pull request by the given commit
+func GetPullRequestByMergedCommit(ctx context.Context, repoID int64, sha string) (*PullRequest, error) {
+	pr := &PullRequest{
+		BaseRepoID:     repoID,
+		MergedCommitID: sha,
```

Comment:
sha is user input ... if it's empty you get a random pull.
Please use proper Where func :)